src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -0,0 +1,105 @@
+package org.broadinstitute.dsde.workbench.sam.service
+
+import akka.http.scaladsl.model.{StatusCode, StatusCodes}
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.sam.WorkbenchException
+import org.broadinstitute.dsde.workbench.sam.directory.JndiDirectoryDAO
+import org.broadinstitute.dsde.workbench.sam.model._
+import org.broadinstitute.dsde.workbench.sam.openam.OpenAmDAO
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * Created by mbemis on 5/22/17.
+  */
+class ResourceService(val openAmDAO: OpenAmDAO, val directoryDAO: JndiDirectoryDAO)(implicit val executionContext: ExecutionContext) extends LazyLogging {
+
+  def listResourceTypes(userInfo: UserInfo): Future[Set[OpenAmResourceType]] = {
+    openAmDAO.listResourceTypes(userInfo)
+  }
+
+  def createResourceType(resourceType: ResourceType, userInfo: UserInfo): Future[OpenAmResourceType] = {
+    val pattern = resourceUrn(resourceType, "*")
+    openAmDAO.createResourceType(resourceType, pattern, userInfo)
+  }
+
+  def updateResourceType(updatedResourceType: OpenAmResourceType, userInfo: UserInfo): Future[OpenAmResourceType] = {
+    openAmDAO.updateResourceType(updatedResourceType, userInfo)
+  }
+
+  def getDefaultPolicySet(userInfo: UserInfo): Future[OpenAmPolicySet] = {
+    openAmDAO.getDefaultPolicySet(userInfo)
+  }
+
+  def updateDefaultPolicySet(updatedPolicySet: OpenAmPolicySet, userInfo: UserInfo): Future[OpenAmPolicySet] = {
+    openAmDAO.updateDefaultPolicySet(updatedPolicySet, userInfo)
+  }
+
+  def hasPermission(resourceType: ResourceType, resourceId: String, action: String, userInfo: UserInfo): Future[Boolean] = {
+    val urn = resourceUrn(resourceType, resourceId)
+    openAmDAO.evaluatePolicy(Set(urn), userInfo).map { policyEval =>
+      val actionsMap = policyEval.map(p => p.resource -> p.actions).toMap
+      actionsMap.getOrElse(urn, throw new WorkbenchException(s"resource ${resourceUrn(resourceType, resourceId)} not found")).getOrElse(action, false)
+    }
+  }
+
+  def createResource(resourceType: ResourceType, resourceId: String, userInfo: UserInfo): Future[Set[OpenAmPolicy]] = {
+    Future.traverse(resourceType.roles) { role =>
+      val roleMembers: Set[SamSubject] = role.roleName match {
+        case resourceType.ownerRoleName => Set(userInfo.userId)
+        case _ => Set.empty
+      }
+      for {
+        group <- directoryDAO.createGroup(SamGroup(SamGroupName(s"${resourceType.name}-${resourceId}-${role.roleName}"), roleMembers))
+        adminUserInfo <- getOpenAmAdminUserInfo
```

Comment:
i put this here as a possibly temporary solution.

do we want to allow All Authenticated Users to manage policies and use their token? or do we want to manage the policies as the admin?